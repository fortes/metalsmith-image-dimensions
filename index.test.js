/* eslint-env jest,node */
const imageDimensions = require('./index');
const Metalsmith = require('metalsmith');
const {promisify} = require('util');
const rimraf = promisify(require('rimraf'));
const path = require('path');

let files;

beforeAll(done => {
  // Mock out `console.error`
  /* eslint-disable no-console */
  const originalConsoleError = console.error;

  console.error = jest.fn(() => {});
  // Setup metalsmith, but don't build yet.
  const metal = Metalsmith(__dirname)
    .source('./')
    .ignore(['.*', 'node_modules', '__*', 'coverage'])
    .destination('__build')
    .use((files, metalsmith, done) => {
      // Manually add fake HTML files for test
      Object.assign(files, {
        // 'no_images.html': {contents: new Buffer('<p>Just prose.</p>')},
        'simple_image.html': {
          contents: new Buffer('<img src=baboon.png>'),
        },
        'multiple_images.html': {
          contents: new Buffer(`<img src=baboon.png>
        <figure><img src=waterfall.jpg></figure>`),
        },
        'relative/path.html': {
          contents: new Buffer('<img src=../tinygif.gif width=10 height=10>'),
        },
        'already_specified.html': {
          contents: new Buffer('<img src=tinygif.gif width=20 height=20>'),
        },
        'external_image.html': {
          contents: new Buffer('<img src=http://httpbin.org/image/jpeg>'),
        },
        '404_image.html': {
          contents: new Buffer('<img src=bogus.jpg>'),
        },
        'image_without_size_metadata.html': {
          contents: new Buffer('<img src=waterfall.webp>'),
        },
      });

      setImmediate(done);
    })
    .use(imageDimensions());

  metal.build((err, processedFiles) => {
    files = processedFiles;
    done(err);
  });
});

afterAll(() => {
  console.error = originalConsoleError;
  return rimraf(path.join(__dirname, '__build'));
});

test('integration test', () => {
  Object.entries(files).forEach(([file, data]) => {
    if (!/\.html$/.test(file)) {
      return;
    }

    expect({file, contents: data.contents.toString()}).toMatchSnapshot();
  });

  console.error.mock.calls.forEach(call => {
    expect(call[0]).toMatchSnapshot();
  })
});
