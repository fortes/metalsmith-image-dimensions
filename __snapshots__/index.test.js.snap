// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test 1`] = `
Object {
  "contents": "<img src=\\"baboon.png\\" width=\\"512\\" height=\\"512\\">",
  "file": "simple_image.html",
}
`;

exports[`integration test 2`] = `
Object {
  "contents": "<img src=\\"baboon.png\\" width=\\"512\\" height=\\"512\\">
        <figure><img src=\\"waterfall.jpg\\" width=\\"200\\" height=\\"150\\"></figure>",
  "file": "multiple_images.html",
}
`;

exports[`integration test 3`] = `
Object {
  "contents": "<img src=../tinygif.gif width=10 height=10>",
  "file": "relative/path.html",
}
`;

exports[`integration test 4`] = `
Object {
  "contents": "<img src=tinygif.gif width=20 height=20>",
  "file": "already_specified.html",
}
`;

exports[`integration test 5`] = `
Object {
  "contents": "<img src=http://httpbin.org/image/jpeg>",
  "file": "external_image.html",
}
`;

exports[`integration test 6`] = `
Object {
  "contents": "<img src=bogus.jpg>",
  "file": "404_image.html",
}
`;

exports[`integration test 7`] = `
Object {
  "contents": "<img src=waterfall.webp>",
  "file": "image_without_size_metadata.html",
}
`;

exports[`integration test 8`] = `"Cannot find file for image: <img src=\\"bogus.jpg\\">"`;

exports[`integration test 9`] = `"No size info image: waterfall.webp. Did you call readImageSizes?"`;
